doc/advanced_data_structures.rst

```diff
@@ -0,0 +1,117 @@
+******************************
+Advanced Data Structure Topics
```

Comment:
I think part of the feedback on the original documentation is that dataset aggregation was really buried in the docs.  I wonder if we should move it to the top level under ``Dataset`` in the left nav bar and do ``Dataset aggregation``

Reply:
Renamed section to "Advanced topics" to match original intent. This will result in these topics being readily accessible as top-level items on the right and thus easy to find.